test/xkeyboard-config-test.py.in

```diff
@@ -1,97 +1,187 @@
 #!/usr/bin/env python3
+
+from __future__ import annotations
+
 import argparse
+import dataclasses
+import itertools
 import multiprocessing
-import sys
-import subprocess
 import os
+import subprocess
+import sys
 import xml.etree.ElementTree as ET
+from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
 from pathlib import Path
-
-
-verbose = False
+from typing import (
+    Any,
+    ClassVar,
+    Generator,
+    Iterable,
+    NoReturn,
+    Protocol,
+    TextIO,
+    TypeVar,
+    cast,
+)
+
+try:
+    # Python >=3.11 only
+    from typing import Self
+except ImportError:
+    Self = Any
+
+WILDCARD = "*"
 
 DEFAULT_RULES_XML = "@XKB_CONFIG_ROOT@/rules/evdev.xml"
 
 # Meson needs to fill this in so we can call the tool in the buildir.
 EXTRA_PATH = "@MESON_BUILD_ROOT@"
-os.environ["PATH"] = ":".join([EXTRA_PATH, os.getenv("PATH")])
-
+os.environ["PATH"] = ":".join(filter(None, (EXTRA_PATH, os.getenv("PATH"))))
```

Comment:
I'm using Python for some time and didn't know `filter(None, ...)` was a thing :)

It might be a bit clearer to do

```py
":".join(path for path in (EXTRA_PATH, os.getenv("PATH")) if path)
```

but the filter is OK too.

Reply:
I change to `bool` instead of `None`.